ntoskrnl/ps/quota.c

```diff
@@ -816,6 +817,32 @@ PsChargeProcessNonPagedPoolQuota(
     return PsChargeProcessPoolQuota(Process, NonPagedPool, Amount);
 }
 
+/**
+ * @brief
+ * Charges the current process non paged pool quota by the given amount.
+ * 
+ * @param[in] Amount
+ * The amount of quotas to charge into the current process.
+ * 
+ * @return NTSTATUS
+ * Returns STATUS_SUCCESS if quota charing has
+ * suceeded, STATUS_QUOTA_EXCEEDED is returned
+ * otherwise to indicate the caller attempted
+ * to charge quotas over the limits.
+ */
+NTSTATUS
+NTAPI
+PsChargeCurrentProcessNonPagedPoolQuota(
```

Comment:
This function does not exist on Windows nor I see any point of this, so please revert.